arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -1040,20 +1079,24 @@ size_t IResearchViewExecutor<ordered, materializeType>::skipAll() {
 
   size_t skipped = 0;
 
-  for (size_t count = this->_reader->size(); _readerOffset < count;) {
-    if (!_itr && !resetIterator()) {
-      continue;
-    }
-
-    while (_itr->next()) {
-      skipped++;
+  if (_readerOffset < this->_reader->size()) {
+    if (this->infos().filterConditionIsEmpty()) {
+      skipped =  this->_reader->live_docs_count();
+      TRI_ASSERT(_totalPos <= skipped);
+      skipped -= std::min(skipped, _totalPos);
+      _readerOffset = this->_reader->size();
+    } else {
```

Comment:
Is it required to set `_currentSegmentPos` to either 0 or increase it by `this->_reader->size()` at the end of this `if` branch, in case there are any places reading `_currentSegmentPos` afterwards?

Reply:
After increasing _readerOffset to this->_reader->size()  executor is "exhausted" so _currentSegmentPos does not matter, but agreed, it is better to be set to 0 as we 'switched' to next segment.